template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/spark/SparkOnYarnRoleConfigProvider.java

```diff
@@ -20,15 +24,23 @@ public class SparkOnYarnRoleConfigProvider extends AbstractRoleConfigProvider {
 
     private static final String SPARK_YARN_ACCESS_DIR_PARAM = "spark.yarn.access.hadoopFileSystems=";
 
+    private static final String SPARK_HADOOP_S3_SSL_CHANNEL_MODE = "spark.hadoop.fs.s3a.ssl.channel.mode=openssl";
+
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case SparkRoles.GATEWAY:
-                return ConfigUtils.getStorageLocationForServiceProperty(source, HMS_METASTORE_EXTERNAL_DIR)
-                        .map(location -> ConfigUtils.getBasePathFromStorageLocation(location.getValue()))
-                        .map(basePath -> List.of(config(SPARK_CONF_CLIENT_SAFETY_VALVE,
-                                SPARK_YARN_ACCESS_DIR_PARAM + basePath)))
-                        .orElseGet(List::of);
+                List<ApiClusterTemplateConfig> roleConfigs = new ArrayList<>();
+                Optional<StorageLocationView> storageLocationView = ConfigUtils.getStorageLocationForServiceProperty(source, HMS_METASTORE_EXTERNAL_DIR);
+                if (storageLocationView.isPresent()) {
+                    roleConfigs.add(config(SPARK_CONF_CLIENT_SAFETY_VALVE,
+                            SPARK_YARN_ACCESS_DIR_PARAM + ConfigUtils.getBasePathFromStorageLocation(storageLocationView.get().getValue())));
+                }
+                if (CMRepositoryVersionUtil.isS3SslChannelModeSupported(source.getBlueprintView()
```

Comment:
Same question here about setting the safety_valve twice in the list..

Reply:
These configurations are delta. They always get prefixed with existing configs from template. Please look at addRoleConfigs in CmTemplateProcessor.